src/disjoint_mut.rs

```diff
@@ -202,12 +205,12 @@ impl<T: AsMutPtr> DisjointMut<T> {
     /// other mutable borrows from this collection overlap with the returned
     /// immutably borrowed region for the lifetime of that borrow.
     ///
-    /// # Safety
-    ///
-    /// Caller must ensure that no elements of the resulting borrowed slice or
-    /// element are concurrently mutably borrowed at all during the lifetime of
-    /// the returned borrow.
-    pub unsafe fn index<'a, I>(
+    /// Caller must take care that no elements of the resulting borrowed slice
+    /// or element are concurrently mutably borrowed at all by [`index_mut()`]
+    /// during the lifetime of the returned borrow. We do not consider this
+    /// method unsafe because it cannot result in a race condition without
+    /// calling [`index_mut()`] with an overlapping range.
```

Comment:
```suggestion
    /// Caller must take care that no elements of the resulting borrowed slice
    /// or element are concurrently mutably borrowed at all by [`Self::index_mut`]
    /// during the lifetime of the returned borrow. We do not consider this
    /// method `unsafe` because it cannot result in a race condition without
    /// calling [`Self::index_mut`] with an overlapping range.
    ///
    /// Furthermore, because this method is not publically accessible outside of this crate,
    /// we can ensure all calls are safe as a whole rather than add a
    /// virtually identical `unsafe {}` block and `// SAFETY` comment for each one.
    /// Those `// SAFETY` comments wouldn't add any useful information,
    /// so we're instead moving the safety comment here and asserting all uses are safe altogether.
```

Reply:
I rewrote the comment, look ok now?